docs/providers.rst

```diff
@@ -0,0 +1,111 @@
+Providers for GCP-related Services
+==================================
+
+.. TODO: externally link "interfaces" to gordon-dns's docs once published.
+
+This package provides the following interface providers. Use any or all of them; they are not dependent on each other.
+
+
+Provider APIs
+-------------
+
+.. toctree::
+   :maxdepth: 1
+
+   compute
+   dns
+   pubsub
+
+
+Example Gordon Configuration with GCP Plugin
+--------------------------------------------
+
+.. literalinclude:: ../gordon.toml.example
+   :language: ini
+
+
+Plugin Configuration
+--------------------
+
+.. _gcp:
+
+``[gcp]``
+~~~~~~~~~
+
+.. program:: gcp
+
+Global configuration for all GCP plugin providers.
+
+.. option:: project = "STR"
+
+    GCP `project`_ for all providers to act within.
+
+.. option:: keyfile = "STR"
+
+    Key needed for all providers to use when interacting with the given ``project``.
+
+
+.. _gcpgce:
+
+``[gcp.compute]``
+~~~~~~~~~~~~~~~~~
+
+.. note::
+
+    Configuration defined for the ``compute`` provider will overwrite values of the same keys defined under :ref:`gcp <gcp>`, then inherit the rest.
+
+    The ``compute`` provider does **not** have access to configuration for :ref:`gcp.dns <gcpdns>` and :ref:`gcp.pubsub <gcppubsub>`.
```

Comment:
just a suggestion but might want to just add a note at the top saying this, rather than saying it in every description?

Reply:
ah yeah good call.